lib/matplotlib/ticker.py

```diff
@@ -730,19 +733,23 @@ def format_data_short(self, value):
                 (math.floor(math.log10(abs(value))) + 1 if value else 1)
                 - math.floor(math.log10(delta)))
             fmt = f"%-#.{sig_digits}g"
-        return (
-            self.fix_minus(
-                locale.format_string(fmt, (value,)) if self._useLocale else
-                fmt % value))
+        return self._format_maybe_minus_and_locale(fmt, value)
 
     def format_data(self, value):
         # docstring inherited
-        if self._useLocale:
-            s = locale.format_string('%1.10e', (value,))
+        e = math.floor(math.log10(value))
+        s = round(value / 10**e, 10)
+        exponent = self._format_maybe_minus_and_locale("%d", e)
+        significand = self._format_maybe_minus_and_locale(
+            "%d" if s % 1 == 0 else "%1.10e", s)
```

Comment:
Is `%e` really the right format now that `s` is divided by the exponent?

Reply:
I don't think it matters?  but I agree %f may be clearer.